packages/clarity-js/src/data/metadata.ts

```diff
@@ -172,6 +189,30 @@ function setCookie(key: string, value: string, time: number): void {
     let expiry = new Date();
     expiry.setDate(expiry.getDate() + time);
     let expires = expiry ? Constant.Expires + expiry.toUTCString() : Constant.Empty;
-    document.cookie = `${key}=${value}${Constant.Semicolon}${expires}${Constant.Path}`;
+    let cookie = `${key}=${value}${Constant.Semicolon}${expires}${Constant.Path}`;
+    try {
+      // Attempt to get the root domain only once and fall back to writing cookie on the current domain.
+      if (rootDomain === null) {
+        let hostname = location.hostname ? location.hostname.split(Constant.Dot) : [];
+        // Walk backwards on a domain and attempt to set a cookie, until successful
+        for (let i = hostname.length - 1; i >= 0; i--) {
+          rootDomain = `.${hostname[i]}${rootDomain ? rootDomain : Constant.Empty}`;
+          // We do not wish to attempt writing cookie on the absolute last part, e.g. .com or .net.
+          // So we start attempting after second-last part, e.g. .domain.com (PASS) or .co.uk (FAIL)
+          if (i < hostname.length - 1) { 
+            // Write the cookie on the current computed top level domain
+            document.cookie = `${cookie}${Constant.Semicolon}${Constant.Domain}${rootDomain}`;
+            // Once written, check if the cookie was set successfully
+            // If yes, no more action required and return from the function. rootDomain is already set
+            // If no, then continue with the for loop
+            if (document.cookie.indexOf(key) >= 0) { return; }
```

Comment:
This might match if the key is a subpart of soe other cookie. Would be cleaner to write a set/getcookie routine somewhere and call those here.

Reply:
Good point. We already have getCookie and using that instead. Less code + safer.